### PR TITLE
Review of cpu tests and creation of a class for mixed cpu tests.

### PR DESCRIPTION
**Pull request by LDj3SNuD**
~~Added a test control system~~ also in compliance with the Setup and Teardown methods that are called for each new Test or TestCase.
Added six mixed tests.

### CODE REVIEW

**Comment by gdkchan** [comment]
FYI @MerryMage

File: Ryujinx.Tests/Cpu/CpuTest.cs [hunk 1]
```diff
@@ -9,65 +9,162 @@
...
+            Memory = null;
+            Allocator = null;
             Marshal.FreeHGlobal(Ram);
         }
 
-        private void Execute(AThread Thread)
+        protected void Opcode(uint Opcode)
+        {
+            if ((Opcode & 0xFFE0001F) != 0xD4200000 && // BRK #<imm>
+                (Opcode & 0xFFE0FC00) != 0xD6400000) // RET {<Xn>}
```
Comment (MerryMage) [inline Ryujinx.Tests/Cpu/CpuTest.cs]: Personally I'd rather not have this.

File: Ryujinx.Tests/Cpu/CpuTest.cs [hunk 2]
```diff
@@ -9,65 +9,162 @@
...
+                {
+                    BrkRetOpcodes();
+                }
 
-            Wait.Reset();
-            Thread.Execute();
-            Wait.WaitOne();
+                if (!SetThreadStateControl)
+                {
+                    SetThreadState();
```
Comment (MerryMage) [inline Ryujinx.Tests/Cpu/CpuTest.cs]: It's not necessary to call `SetThreadState()` here in any case.

File: Ryujinx.Tests/Cpu/CpuTest.cs [hunk 3]
```diff
@@ -9,65 +9,162 @@
...
-            AutoResetEvent Wait = new AutoResetEvent(false);
-            Thread.ThreadState.Break += (sender, e) => Thread.StopExecution();
-            Thread.WorkFinished += (sender, e) => Wait.Set();
+            if (!ExecuteOpcodesControl)
+            {
+                ExecuteOpcodesControl = true;
+
+                if (!BrkRetOpcodesControl)
+                {
+                    BrkRetOpcodes();
```
Comment (MerryMage) [inline Ryujinx.Tests/Cpu/CpuTest.cs]: imo if a test requires these it should call this itself.

**Comment by MerryMage** [comment]
Are `BrkRetOpcodesControl`, `SetThreadStateControl`, `ExecuteOpcodesControl`, `SingleOpcodeControl` truly necessary? There's no real need to restrict these to call-once.

In any case, if that was truly an error, surely Assert.Ignore would not be appropriate?

LGTM otherwise.

**Comment by LDj3SNuD** [comment]
Changes made in compliance with the requests of the reviewer and NUnit.
Modified a mixed test.

**Pull request merged**